### PR TITLE
Fix stage backend

### PR DESCRIPTION
**Pull request by zhu327**
### Description

Fixes # (issue)

### Checklist

- [ ] 填写 PR 描述及相关 issue (write PR description and related issue)
- [ ] 代码风格检查通过 (code style check passed)
- [ ] PR 中包含单元测试 (include unit test)
- [ ] 单元测试通过 (unit test passed)
- [ ] 本地开发联调环境验证通过 (local development environment verification passed)

### CODE REVIEW

**Review by alex-smile — APPROVED** [review]

File: src/dashboard/apigateway/apigateway/core/management/commands/migrate_backend.py [hunk 1]
```diff
@@ -88,54 +89,70 @@ def _handle_gateway(self, gateway: Gateway):
...
+                backend_stage_config[backend.id] = resource_stage_config
 
-        # 清理未使用的backend
-        used_backend_ids = list(
-            Proxy.objects.filter(resource__gateway=gateway).values_list("backend_id", flat=True).distinct()
-        )
-        used_backend_ids.append(default_backend.id)
+    def _get_max_resource_backend_count(self, gateway: Gateway):
+        count = 0
+        backends = Backend.objects.filter(gateway=gateway, name__startswith="backend-").only("name")
```
Comment (wklken) [inline src/dashboard/apigateway/apigateway/core/management/commands/migrate_backend.py:109]: 直接value_list flat=True?

File: src/dashboard/apigateway/apigateway/core/management/commands/migrate_backend.py [hunk 2]
```diff
@@ -150,33 +167,28 @@ def _handle_resource_backend(
                         _host = _host.replace("{env." + key + "}", vars[key])
 
                 hosts.append({"scheme": scheme, "host": _host, "weight": host["weight"]})
+            hosts.sort(key=lambda x: "{}://{}#{}".format(x["scheme"], x["host"], x["weight"]))
```
Comment (wklken) [inline src/dashboard/apigateway/apigateway/core/management/commands/migrate_backend.py:170]: sort在两个地方用到了, 提取独立函数避免可能得漏改
需要注释sort的目的

File: src/dashboard/apigateway/apigateway/core/management/commands/migrate_backend.py [hunk 3]
```diff
@@ -150,33 +167,28 @@ def _handle_resource_backend(
...
-                backend=backend,
-                stage=stage,
-                config=config,
-            )
-            backend_configs.append(backend_config)
+        return stage_config
 
-        if backend_configs:
-            BackendConfig.objects.bulk_create(backend_configs)
+    def _gen_backend_config_map(self, gateway: Gateway) -> Dict[int, Dict[int, Any]]:
```
Comment (wklken) [inline src/dashboard/apigateway/apigateway/core/management/commands/migrate_backend.py:181]: _get_backend_stage_config_map

File: src/dashboard/apigateway/apigateway/core/management/commands/migrate_backend.py (hunk 1 above)
Comment (zhu327) [reply src/dashboard/apigateway/apigateway/core/management/commands/migrate_backend.py]: Done

File: src/dashboard/apigateway/apigateway/core/management/commands/migrate_backend.py (hunk 2 above)
Comment (zhu327) [reply src/dashboard/apigateway/apigateway/core/management/commands/migrate_backend.py]: Done

File: src/dashboard/apigateway/apigateway/core/management/commands/migrate_backend.py (hunk 3 above)
Comment (zhu327) [reply src/dashboard/apigateway/apigateway/core/management/commands/migrate_backend.py]: Done

**Review by wklken — APPROVED** [review]
LGTM

**Pull request merged**